internal/handlers/handlers.go

```diff
@@ -57,18 +30,31 @@ func (h *Handler) ShortenHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Could not shorten URL", http.StatusBadRequest)
 		return
 	}
+	responseURL := h.cfg.BaseURL + "/" + shortURL
 
-	// write shortened URL
-	w.WriteHeader(http.StatusCreated)
-	if _, err := w.Write([]byte(h.cfg.BaseURL + "/" + shortURL)); err != nil {
-		http.Error(w, "Could not write URL", http.StatusInternalServerError)
-		return
-	}
-
-	// append entry uuid - shorturl - longurl to file
+	// check existing shortURL
 	err = h.st.AddNewEntry(shortURL, longURL)
 	if err != nil {
+		var urlConflictError *t.URLConflictError
+		if errors.As(err, &urlConflictError) { // get instance of URLConflictError if err matches
+			if urlConflictError.ShortURL != "" {
+				responseURL = h.cfg.BaseURL + "/" + urlConflictError.ShortURL
+			} else {
+				responseURL = h.cfg.BaseURL + "/" + shortURL
+			}
+			w.WriteHeader(http.StatusConflict)
+		} else {
+			http.Error(w, "Could not write URL to storage", http.StatusInternalServerError)
+			return
+		}
+	} else {
+		w.WriteHeader(http.StatusCreated)
+	}
```

Comment:
Мне кажется отсутствие `else` там, где его можно опустить улучшает читаемость кода.
```suggestion
		var urlConflictError *t.URLConflictError
		if errors.As(err, &urlConflictError) { // get instance of URLConflictError if err matches
			if urlConflictError.ShortURL != "" {
				responseURL = h.cfg.BaseURL + "/" + urlConflictError.ShortURL
			} else {
				responseURL = h.cfg.BaseURL + "/" + shortURL
			}
			w.WriteHeader(http.StatusConflict)
		}
		http.Error(w, "Could not write URL to storage", http.StatusInternalServerError)
		return
	}
	w.WriteHeader(http.StatusCreated)
```

Reply:
мне кажется, w.WriteHeader(http.StatusCreated) перезапишет w.WriteHeader(http.StatusConflict)
но суть понял